src/components/HistoryTable/HistoryTable.tsx

```diff
@@ -35,10 +37,37 @@ const HistoryTable: React.FC<HistoryTableProps> = ({
   }, [recentlyFetchedCommits]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    if (listRef.current) {
-      listRef.current.resetAfterIndex(0);
+    if (processedCommitCount > 0) {
+      if (listRef.current) {
+        listRef.current.resetAfterIndex(0);
+      }
+    }
+  }, [processedCommitCount]);
+
+  useEffect(() => {
+    if (selectedCommit) {
+      if (
+        selectedCommit.loaded &&
+        !selectedCommit.visited &&
+        selectedCommit.rowIndex
+      ) {
+        if (listRef.current) {
+          listRef.current.scrollToItem(selectedCommit.rowIndex, "center");
+          markSelectedVisited();
+        }
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedCommit?.loaded]);
```

Comment:
if the rowIndex changes, we should scroll to it. We don't actually need visited or loaded if we look at all commits to see if we can find the selected commits rowIndex.

Reply:
Great point!